Adds info about new codecs (#4571)
opensearch-project/documentation-website: _tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore.md

Request fields | Description
:--- | :---
`indices` | The indexes you want to include in the snapshot. You can use `,` to create a list of indexes, `*` to specify an index pattern, and `-` to exclude certain indexes. Don't put spaces between items. Default is all indexes.

> `ignore_unavailable` | If an index from the `indices` list doesn't exist, whether to ignore it rather than fail the snapshot. Default is false.

Author: ```suggestion
`ignore_unavailable` | If an index from the `indices` list doesn't exist, whether to ignore it rather than fail the snapshot. Default is `false`.
```

> `include_global_state` | Whether to include cluster state in the snapshot. Default is true.

Author: ```suggestion
`include_global_state` | Whether to include cluster state in the snapshot. Default is `true`.
```

> `partial` | Whether to allow partial snapshots. Default is false, which fails the entire snapshot if one or more shards fails to store.

Author: ```suggestion
`partial` | Whether to allow partial snapshots. Default is `false`, which fails the entire snapshot if one or more shards fails to store.
```

Request parameters | Description
:--- | :---

> `indices` | The indexes you want to restore. You can use `,` to create a list of indexes, `*` to specify an index pattern, and `-` to exclude certain indexes. Don't put spaces between items. Default is all indexes.

Reviewer: Entire list: Should the defaults be in tic marks?

Author: Yes, looks like we add tic marks to the default values.

We recommend ceasing write requests to a cluster before restoring from a snapshot, which helps avoid scenarios such as:

1. You delete an index, which also deletes its alias.
1. A write request to the now-deleted alias creates a new index with the same name as the alias.
1. The alias from the snapshot fails to restore due to a naming conflict with the new index.

Snapshots are only forward-compatible by one major version. If you have an old snapshot, you can sometimes restore it into an intermediate cluster, reindex all indexes, take a new snapshot, and repeat until you arrive at your desired version, but you might find it easier to just manually index your data on the new cluster.

When creating a snapshot, the index codec setting influences both the size of the snapshot and the time required for its creation. If the codec of an index is updated, new snapshots created will utilize the latest codec setting. The resulting snapshot size will reflect the compression characteristics of the latest codec settings. Existing segments included in the snapshot retain their original compression characteristics. 

> If you want to restore the indexes from a snapshot of a cluster to another cluster, then it is important to verify that the target cluster supports the codecs of the segments present in the source snapshot. For example, if the source snapshot has the segments of `zstd` or `zstd_no_dict` codecs (introduced in OpenSearch 2.9), then the user cannot restore to the target cluster where the support of these codecs is not available. 

Reviewer 2: ```suggestion
If you want to restore the indexes from a snapshot of a cluster to another cluster, it is important to verify that the target cluster supports the codecs used for the segments present in the source snapshot. For example, if the source snapshot uses the `zstd_no_dict` codec with its segments (introduced in OpenSearch 2.9), the user cannot restore to the target cluster unless the target cluster supports this codec. 
```

- To perform snapshot and restore operations, users must have the built-in `manage_snapshots` role.

> - You can't restore snapshots that contain global state or the `.opendistro_security` index.

Reviewer: ```suggestion
- You can't restore snapshots that contain a global state or the `.opendistro_security` index.
```

> If a snapshot contains global state, you must exclude it when performing the restore. If your snapshot also contains the `.opendistro_security` index, either exclude it or list all the other indexes you want to include:

Reviewer: ```suggestion
If a snapshot contains a global state, you must exclude it when performing the restore. If your snapshot also contains the `.opendistro_security` index, either exclude it or list all the other indexes you want to include:
```
